lib/publiccloud/instance.pm

```diff
@@ -976,4 +976,40 @@ sub wait_for_state {
     die("The instance state is not '$state' but '$current' instead.");
 }
 
+sub check_selinux_pc {
+    # Check SELinux-enforced VMs on publiccloud
+    my ($self) = @_;
+    my $log_avc = "selinux_avc_audit.txt";
+    my $ssh_opts = "-t -o ControlPath=none";
+    die("Unexpected environment.") unless (is_public_cloud && (is_sle_micro || is_sle));
+
+    my $instance = $self;    # $self->{my_instance};
+    my $out;
+
+    $out = $instance->ssh_script_output("cat /sys/kernel/security/lsm", ssh_opts => $ssh_opts);
+    die("unexpected SELinux status in kernel security lsm: $out") unless (grep(/^selinux$/, split(/,/, $out)));
+    record_info("SELinux status", $out);
+
+    $out = $instance->ssh_script_output("sestatus", ssh_opts => $ssh_opts);
+    die("Unexpected SELinux status: $out") unless (grep(/Current mode:\s+enforcing/, split(/\n+/, $out)));
+    record_info("SELinux status", $out);
+
+    # Check audit status:
+    $instance->ssh_assert_script_run("sudo systemctl is-active auditd", ssh_opts => $ssh_opts);
+    $instance->ssh_assert_script_run("sudo [ -s /var/log/audit/audit.log ]", ssh_opts => $ssh_opts);
+    $instance->ssh_assert_script_run("sudo which ausearch", ssh_opts => $ssh_opts);
```

Comment:
You can drop this redundant check given that you run this command immediately afterwards.

```suggestion
```